### PR TITLE
Improve documentation styling

### PR DESCRIPTION
**Pull request by gautierronan**
This PR introduces several styling modifications to the documentation. The overall goal is to make it look more clean.

- Smaller fonts overall
- Removes orange left border on code snippets (but they stay on function signatures in the API)
- Increase contrast in dark mode
- Improve page spacing
- Change admonition styling (only left border now)
- Remove "quotes" admonition altogether and associated CSS
- Add titles to all notes
- Fold all notes by default

For the latter, the reasoning is that if it's a note, it's not that important and does not need to appear right away. Otherwise it should be probably be in the main text or inside a warning.

To review, I advise moving around the documentation tutorials and API reference with a `task docserve`, on top of reading the code diff.

### CODE REVIEW

**Review by abocquet — APPROVED** [review]

File: docs/tutorials/time-dependent-operators.md [hunk 1]
```diff
@@ -238,7 +238,7 @@ Array([[1., 0.],
 !!! Warning "The function `f` must return a JAX array (not an array-like object!)"
     An error is raised if the function `f` does not return a JAX array. This error concerns any other array-like objects. This is enforced to avoid costly conversions at every time step of the numerical integration.
 
-??? Notes "Batching"
+??? Notes "Batching callable time-arrays"
```
Comment (pierreguilmin) [inline docs/tutorials/time-dependent-operators.md:241]:
```suggestion
??? Notes "Batching arbitrary time-dependent operators"
```

File: dynamiqs/mesolve/mesolve.py [hunk 2]
```diff
@@ -56,15 +56,15 @@ def mesolve(
     where $H(t)$ is the system's Hamiltonian at time $t$ and $\{L_k(t)\}$ is a
     collection of jump operators at time $t$.
 
-    Quote: Time-dependent Hamiltonian or jump operators
+    Notes-: Defining a time-dependent Hamiltonian or jump operator
```
Comment (pierreguilmin) [inline dynamiqs/mesolve/mesolve.py:59]:
```suggestion
    Notes-: Defining a time-dependent Hamiltonian or jump operators
```

File: dynamiqs/plots/plots_wigner.py [hunk 3]
```diff
@@ -264,7 +264,7 @@ def plot_wigner_gif(
         This function creates files in the current working directory under
         `.tmp/dynamiqs`.
 
-    Note:
+    Notes-: Display in Jupyter
```
Comment (pierreguilmin) [inline dynamiqs/plots/plots_wigner.py:267]:
```suggestion
    Notes-: Display in Jupyter notebooks
```

File: dynamiqs/result.py [hunk 4]
```diff
@@ -158,7 +158,7 @@ class MEResult(Result):
         - If the option `cartesian_batching = True` (default value), the results leading
           dimensions are
           ```
-          ... = ...H, ...L0, ...L1, (...), ...rho0
+          ... = ...H, ...L0, ...L1, ...Ln, ...rho0
```
Comment (pierreguilmin) [inline dynamiqs/result.py:161]: I would revert this, it's incorrect.

File: dynamiqs/time_array.py [hunk 5]
```diff
@@ -46,24 +46,24 @@ def pwc(times: ArrayLike, values: ArrayLike, array: ArrayLike) -> PWCTimeArray:
...
         times _(array_like of shape (N+1,))_: Time points $t_k$ defining the boundaries
             of the time intervals, where _N_ is the number of time intervals.
         values _(array_like of shape (..., N))_: Constant values $c_k$ for each time
             interval.
         array _(array_like of shape (n, n))_: Constant array $O_0$.
 
+    Notes-: `times` sorting
+        The argument `times` argument must be sorted in ascending order, but does not
+        need to be evenly spaced.
+
```
Comment (pierreguilmin) [inline dynamiqs/time_array.py:59]: Nothing should be in-between Args and Returns. Typically we put all notes on top of args, and I think it's better not to mix the function description and the args/returns section.

File: dynamiqs/time_array.py [hunk 6]
```diff
@@ -46,24 +46,24 @@ def pwc(times: ArrayLike, values: ArrayLike, array: ArrayLike) -> PWCTimeArray:
...
-
     Args:
         times _(array_like of shape (N+1,))_: Time points $t_k$ defining the boundaries
             of the time intervals, where _N_ is the number of time intervals.
         values _(array_like of shape (..., N))_: Constant values $c_k$ for each time
             interval.
         array _(array_like of shape (n, n))_: Constant array $O_0$.
 
+    Notes-: `times` sorting
+        The argument `times` argument must be sorted in ascending order, but does not
```
Comment (pierreguilmin) [inline dynamiqs/time_array.py:57]:
```suggestion
        The argument `times` must be sorted in ascending order, but does not
```

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
Very nice!

> Smaller fonts overall

In think we should revert this, I already found the font to be borderline small, now it's getting hard to read. Maybe I just need glasses 👓

> Add titles to all notes
> Fold all notes by default
> For the latter, the reasoning is that if it's a note, it's not that important and does not need to appear right away. Otherwise it should be probably be in the main text or inside a warning.

It's a bit annoying to click and open all notes one by one. To me notes != "not that important", it's just a way to add colours to the doc, and separate bits of informations, unrelated to the importance or not of the information. For example in https://www.dynamiqs.org/tutorials/open-systems.html examples have a clear separated styling, and https://www.dynamiqs.org/python_api/utils/utils/lindbladian.html the note is just an extra information that pops out as such. I think folding notes actually _forces_ the curious user to open them (to know what's inside), while he can quickly skim through otherwise. It's annoying to have to click to reveal one line of text. IMO we should only fold very long or very specific notes.

Moreover, when the note is not folded the title is often redundant with the content, and I like keeping "Note" then, it unifies a bit all these blue things throughout the documentation. So I think most of the newly added titles are just redundant informations.

File: dynamiqs/result.py (hunk 4 above)
Comment (pierreguilmin) [reply dynamiqs/result.py:161]: (`n` is typically used for the Hilbert space dimension, and this does not clearly explain that there are arbitrarily many jump operators)

Comment (gautierronan) [reply dynamiqs/result.py:161]: Ok, I didn't really like the previous notation, but I agree this one is not good. I'm reverting for lack of something better.

File: dynamiqs/mesolve/mesolve.py (hunk 2 above)
Comment (pierreguilmin) [reply dynamiqs/mesolve/mesolve.py:59]: Right sorry

File: dynamiqs/plots/utils.py [hunk 7]
```diff
@@ -90,7 +90,7 @@ def gridplot(
     Warning:
         Documentation redaction in progress.
 
-    Notes:
+    Notes-: Shortcut to Matplotlib `plt.subplots()`
```
Comment (pierreguilmin) [inline dynamiqs/plots/utils.py:93]: I would expand this.

**Review by pierreguilmin — CHANGES_REQUESTED** [review]
Beautiful thanks a lot for all the fixes! Minor neat. Also should we stick with `Note` or `Notes`? We still use both.

**Comment by gautierronan** [comment]
> Also should we stick with Note or Notes? We still use both.

So not sure on this one. I prefer "Note" to "Notes" on the documentation when there's no title (it's similar to "Warning") because there's usually a single comment. However, in docstrings, other sections are usually plural ("Args", "Returns", etc.). 

I can either unify everything to be "Note" instead of "Notes", or write "Notes: Note" in docstrings, which sounds kinda dumb haha.

**Comment by pierreguilmin** [comment]
> > Also should we stick with Note or Notes? We still use both.
> 
> So not sure on this one. I prefer "Note" to "Notes" on the documentation when there's no title (it's similar to "Warning") because there's usually a single comment. However, in docstrings, other sections are usually plural ("Args", "Returns", etc.).
> 
> I can either unify everything to be "Note" instead of "Notes", or write "Notes: Note" in docstrings, which sounds kinda dumb haha.

Let's put "Note" everywhere!

**Review by pierreguilmin — APPROVED** [review]
Beautiful 🎨✨

**Pull request merged**